metaflow/plugins/aip/argo_utils.py

```diff
@@ -41,6 +63,52 @@ def run_argo_workflow(
     return argo_run_id, argo_run_uid
 
 
+def get_latest_workflow(
+    kubernetes_namespace: str,
+    project_name: Optional[str] = None,
+    branch_name: Optional[str] = None,
+    template_name_prefix: Optional[str] = None,
+):
+    # TODO:
+    # - Add filter by project_id instead of project name - project_id is not added as a label yet.
+    # - Add filter by flow_name - flow_name is not added as a label yet.
+    client = ArgoClient(namespace=kubernetes_namespace)
+
+    templates = client.list_workflow_template()["items"]
+    templates = [
+        template
+        for template in templates
+        if (
+            not project_name
+            or template["metadata"]["labels"]["gitlab.zgtools.net/project-name"]
+            == project_name
+        )
+        and (
+            not branch_name
+            or template["metadata"]["labels"]["gitlab.zgtools.net/branch-name"]
+            == branch_name
+        )
+        and (
+            not template_name_prefix
+            or template["metadata"]["name"].startswith(template_name_prefix)
+        )
+    ]
+    if not templates:
+        raise AIPException(
+            f"No workflow template found with constraints "
+            f"project_name={project_name}, branch_name={branch_name}, template_name_prefix={template_name_prefix}"
+        )
+    # Sort by creation timestamp to get the latest template.
+    templates.sort(
+        key=lambda template: template["metadata"]["creationTimestamp"], reverse=True
+    )
+    template_name = templates[1]["metadata"]["name"]
```

Comment:
why `templates[1]` and not `templates[0]` to get latest?

Reply:
Nice catch!